addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -2663,6 +2674,302 @@ export class PosStore extends WithLazyGetterTrap {
         });
         await validation.validateOrder(false);
     }
+
+    handlePreparationHistory(srcPrep, destPrep, srcLine, destLine, qty) {
+        const srcKey = srcLine.preparationKey;
+        const destKey = destLine.preparationKey;
+        const srcQty = srcPrep[srcKey]?.quantity;
+
+        if (srcQty) {
+            if (srcQty <= qty) {
+                const newPrep = { ...srcPrep[srcKey], uuid: destLine.uuid };
+                destPrep[destKey] = newPrep;
+                delete srcPrep[srcKey];
+            } else {
+                srcPrep[srcKey].quantity = srcQty - qty;
+                destPrep[destKey] = { ...srcPrep[srcKey], uuid: destLine.uuid, quantity: qty };
+            }
+        }
+    }
+
+    get isSelectedLineCombo() {
+        const selectedOrderline = this.selectedOrder.getSelectedOrderline();
+        return selectedOrderline && selectedOrderline.isPartOfCombo();
+    }
+
```

Comment:
```suggestion
    /**
     * This method is called in two different contexts.
     *
     * 1. Each time `order_summary` is rendered, this method is called in “limited” mode to
     *    determine whether it is possible to create combos with the order lines.
     *
     * 2. When you click on “apply combos,” a popup opens if there are several possibilities, this
     *    time in “full” mode.
     *
     * Limited mode: this mode returns when more than one combo possibility is
     * found.
     *
     * Full mode: returns all possibilities; this mode is more complex, hence the limited mode for
     * rendering.
     */
```

Even if already explained in the code, is always good to have this